qmf/src/plugins/messageservices/imap/imapstrategy.cpp

```diff
@@ -2692,6 +2709,9 @@ void ImapSynchronizeAllStrategy::processUidSearchResults(ImapStrategyContextBase
         qWarning() << "Unable to update folder for account:" << context->config().id();
     }
 
+    // Check if any of the message is flagged as deleted on server side and not expunged yet
+    QStringList flaggedAsRemovedUids = flaggedAsDeletedUids(context);
```

Comment:
This is a confusing naming scheme - does 'removed' imply a different meaning here than 'deleted'?